tools/update_certs.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+# Copyright 2024 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tool for updating Cobalt SSL Certificates.
+
+Usage:
+  update_certs.py -r <cobalt_repo> -b <openssl_bin> -p <pem_files>
+
+Args:
+    cobalt_repo: The path to the Cobalt repository root.
+    openssl_bin: The path to the OpenSSL binary, defaults to "/usr/bin/openssl"
+      when not specified.
+    pem_files: The paths to the source pem files.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+
+_CERTS_PATH = 'cobalt/content/ssl/certs'
+_BUILD_PATH = 'cobalt/network/certs.gni'
+
+
+def SplitCerts(concatenated_cert_file):
+  """Split concatenated .pem file into a list of individual certs."""
+  begin_anchor = '-----BEGIN CERTIFICATE-----'
+  end_anchor = '-----END CERTIFICATE-----'
+
+  cert_list = []
+  current_cert = ''
+  in_cert = False
+
+  for line in concatenated_cert_file:
+    if not in_cert and line.find(begin_anchor) != -1:
+      in_cert = True
+      current_cert += line
+    elif in_cert and line.find(end_anchor) != -1:
+      in_cert = False
+      current_cert += line
+      cert_list.append(current_cert)
+      current_cert = ''
+    elif in_cert:
+      current_cert += line
+
+  return cert_list
+
+
+def ComputeCertFilename(openssl_bin, cert):
+  """Use x509 to get the hashed filename for the individual cert."""
+  command_args = [openssl_bin, 'x509', '-subject_hash', '-noout']
+
+  with subprocess.Popen(
```

Comment:
Can this just be a check_output instead of a Popen?

Reply:
This is copied from the previous script. I tried messing with check_output but the openssl_bin expects to be passed a cert file path and not the contents of the file. I think the use of Popen and communicate was done to pipe in the contents of the file. I can dig further if needed, but I only spent a few minutes looking into it before moving on.

Reply:
Instead of using `communicate` to pass the input, I believe you can use `check_output` with the `input=` kwarg

Reply:
done.